### PR TITLE
hotfix: prevent user-supplied rkey on posts with createRecord

### PR DESCRIPTION
**Pull request by DavidBuchanan314**
Heads up, this PR is completely untested, but this seems like the quickest way to return to sanity

image

### CODE REVIEW

File: packages/pds/src/api/com/atproto/repo/createRecord.ts [hunk 1]
```diff
@@ -33,6 +33,11 @@ export default function (server: Server, ctx: AppContext) {
           'Unvalidated writes are not yet supported.',
         )
       }
+      if (collection === ids.AppBskyFeedPost && rkey !== undefined) {
```
[the remark this answers is not in the document]
Comment (DavidBuchanan314) [reply packages/pds/src/api/com/atproto/repo/createRecord.ts:36]: Good point - I wasn't sure if an empty string would get treated as a literal rkey of length zero

**Review by dholms — APPROVED** [review]
yup yup 👍

**Comment by QuantumFractal** [comment]
It was fun while it lasted 😅

**Comment by DavidBuchanan314** [comment]
It just occurred to me that you probably want to catch this in `applyWrites` too...

**Pull request merged**